macrosynergy/management/dq.py

```diff
@@ -261,6 +266,33 @@ def _fetch(self, endpoint: str = "/groups", select: str = "groups",
 
         return results
 
+    def _fetch_threading(self, endpoint, params: dict):
+
+        url = self.base_url + endpoint
+        select = "instruments"
+
+        results = []
+        while True:
+            with requests.get(url=url, cert=(self.crt, self.key), headers=self.headers,
+                              params=params) as r:
+                last_response = r.text
+
+            response = json.loads(last_response)
+            results.extend(response[select])
+
+            assert "next" in response['links'][1].keys(), \
+                f"'next' missing from links keys: " \
+                f" {response['links'][1].keys()}"
+
+            if response["links"][1]["next"] is None:
+                break
+
+            url = f"{self.base_url:s}{response['links'][1]['next']:s}"
+            params = {}
+
+        print(f"Current Thread: {threading.current_thread()}.")
```

Comment:
Do we need to remove this from the final version:

```python
print(f"Current Thread: {threading.current_thread()}.")
```

Reply:
Nope - used for testing.

Reply:
Removed.